integration-tests/test_unregister.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_unregister(insights_client):
+    """Test insights-client --unregister after registering it
+
+    If the unregistration is successful, `unregister` method returns True
+    """
+    insights_client.register()
+    assert insights_client.is_registered
+
+    registration_status = insights_client.run("--unregister")
+    assert (
+        "Successfully unregistered from the Red Hat Insights Service"
+        in registration_status.stdout
+    )
+    assert not insights_client.is_registered
+
+
+def test_unregister_twice(insights_client):
+    """Test insights-client --unregister on unregistered system
+
+    If the unregistration is successful, `--unregister` returns True on first attempt and
+    false on subsequent attempts.
+    """
+    insights_client.register()
+    assert insights_client.is_registered
+
+    # unregister once
+    registration_status = insights_client.run("--unregister")
+    assert not insights_client.is_registered
+    assert (
+        "Successfully unregistered from the Red Hat Insights Service"
+        in registration_status.stdout
+    )
+    # unregister twice
+    registration_status = insights_client.run("--unregister", check=False)
```

Comment:
maybe `s/registration_status/unregistration_status/` for clarity?

Reply:
registration status is status of system, I mean status of system. this can be true/false.

Reply:
well, in this case it is the status of the unregistration; what you describe would be the output of `--status`